File: 120-triangle/triangle.py
class Solution:
    def minimumTotal(self, triangle: List[List[int]]) -> int:
        ans = 0
        dp = triangle

        for i in range(len(triangle)-2,-1,-1):
            for j in range(len(triangle[i])-1,-1,-1):
                dp[i][j] = triangle[i][j] + min(triangle[i+1][j],triangle[i+1][j+1])

        return dp[0][0]

        

File: 120-triangle/triangle.cpp
class Solution {
public:
    int minimumTotal(vector<vector<int>>& triangle) 
    {
        int n = triangle.size();

        vector<vector<int>> dp = triangle;

        int ans = 0;

        for(int i=n-2 ; i >=0 ; i-- )
        {
            int m = triangle[i].size();

            for(int j = m-1 ; j >=0 ;--j)
            {
                dp[i][j] = triangle[i][j] + min(dp[i+1][j],dp[i+1][j+1]);
            }
        }

        return dp[0][0];
        
    }
};